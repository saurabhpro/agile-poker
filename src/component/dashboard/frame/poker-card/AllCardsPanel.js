import React, { useState, useEffect } from 'react';
import { CardDeck } from 'react-bootstrap';

import Result from './Result';
import PokerCard from './PokerCard';

import database from '../../../Firebase';
import sessionStoreUserName from '../../../utils/sessionStore';

import './cards.css';

export default function AllCardsPanel() {
  const userName = sessionStoreUserName();
  const [cardValues, setCardValues] = useState([]);
  const [selectedSP, setSelectedSP] = useState(undefined);
  const [showResult, setShowResult] = useState(false);

  // cardValues: [1, 2, 3, 5, 8, 13, 20, Infinity],
  // selectedSP: undefined,

  useEffect(() => {
    const unsubscribe = database
      .collection('points')
      .onSnapshot((snapshot) => {
        const sp = snapshot.docs.map((point) => point.data().numeric);
        setCardValues(sp[0]);
      });

    return () => {
      // this is the cleanup of listeners
      unsubscribe(); //it'll detach the listeners
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const lockStoryPointCard = (value) => {
    setSelectedSP(value);

    // Add a new document in collection "result/task/users" with ID 'userName'
    const res = database
      .collection('result')
      .doc('IND-01')
      .collection('users')
      .doc(userName)
      .set({ storyPoint: value });

    console.log('Stored Result: ', res);

    setShowResult(true);
    setCardValues([value]);
  };

  const getStyle = (value) => {
    return {
      width: '10rem',
      height: '15em',
      disabled: selectedSP !== value,
    };
  };

  const getAllCards = () => {
    const cards = [];

    cardValues
      .sort((a, b) => a - b)
      .forEach((element) => {
        cards.push(
          <PokerCard
            key={element}
            value={element}
            style={getStyle(element)}
            onClick={lockStoryPointCard}
          />,
        );
      });

    return cards;
  };

  return (
    <div>
      {!showResult ? (
        <CardDeck
          className="cards"
          style={{
            flexDirection: 'row',
            marginTop: '10px',
            marginBottom: '20px',
          }}
        >
          {getAllCards()}
        </CardDeck>
      ) : (
        <Result />
      )}
    </div>
  );
}
