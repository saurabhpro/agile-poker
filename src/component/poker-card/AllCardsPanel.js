import React, { useState, useEffect } from 'react';
import { Container, CardDeck } from 'react-bootstrap';
import PokerCard from './PokerCard';
import Firebase from '../Firebase';

export default function AllCardsPanel() {
  const [cardValues, setCardValues] = useState([]);
  const [selectedSP, setSelectedSP] = useState(undefined);

  // cardValues: [1, 2, 3, 5, 8, 13, 20, Infinity],
  //   selectedSP: undefined,

  useEffect(() => {
    new Firebase().database
      .collection('points')
      .onSnapshot((snapshot) => {
        const sp = snapshot.docs.map((point) => point.data().numeric);
        setCardValues(sp[0]);
      });
  }, []);

  const lockStoryPointCard = (value) => {
    setSelectedSP(value);
    setCardValues([value]);
  };

  const getStyle = (value) => {
    return {
      width: '10rem',
      height: '15em',
      disabled: selectedSP !== value,
    };
  };

  const getAllCards = () => {
    const cards = [];
    console.log(cardValues);

    cardValues
      .sort((a, b) => a - b)
      .forEach((element) => {
        cards.push(
          <PokerCard
            key={element}
            value={element}
            style={getStyle(element)}
            onClick={lockStoryPointCard}
          />,
        );
      });

    return cards;
  };

  return (
    <Container>
      <CardDeck>{getAllCards()}</CardDeck>
    </Container>
  );
}
