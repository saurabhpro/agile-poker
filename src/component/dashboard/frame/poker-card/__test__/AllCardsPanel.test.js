import React from 'react';
import { shallow } from 'enzyme';
import AllCardsPanel from '../AllCardsPanel';

let component;

describe('cards with story points panel', () => {
  beforeAll(() => {
    component = shallow(<AllCardsPanel />);
    //console.debug("This was prited",component.debug());
  });

  it('matches the rendered snapshot', () => {
    expect(component).toMatchSnapshot();
  });


  it('should render one <CardDeck>', () => {
    expect(component.find('CardDeck')).toHaveLength(1);
  });
});
