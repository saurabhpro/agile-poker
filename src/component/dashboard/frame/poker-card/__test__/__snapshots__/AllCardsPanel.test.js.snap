// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cards with story points panel matches the rendered snapshot 1`] = `
<div>
  <CardDeck
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": "20px",
        "marginTop": "10px",
      }
    }
  />
</div>
`;
